lib/charms/tls_certificates_interface/v1/tls_certificates.py

```diff
@@ -272,1089 +22,39 @@ def _on_certificate_revoked(self, event: CertificateRevokedEvent) -> None:
-        ]
-        for certificate_signing_request in requirer_unit_csrs:
-            if certificate_signing_request not in provider_csrs:
-                self.on.certificate_creation_request.emit(
-                    certificate_signing_request=certificate_signing_request,
-                    relation_id=event.relation.id,
-                )
-        self._revoke_certificates_for_which_no_csr_exists(relation_id=event.relation.id)
-
-    def _revoke_certificates_for_which_no_csr_exists(self, relation_id: int) -> None:
-        """Revokes certificates for which no unit has a CSR.
-
-        Goes through all generated certificates and compare agains the list of CSRS for all units
-        of a given relationship.
-
-        Args:
-            relation_id (int): Relation id
-
-        Returns:
-            None
-        """
-        certificates_relation = self.model.get_relation(
-            relation_name=self.relationship_name, relation_id=relation_id
-        )
-        if not certificates_relation:
-            raise RuntimeError(f"Relation {self.relationship_name} does not exist")
-        provider_relation_data = _load_relation_data(certificates_relation.data[self.charm.app])
-        list_of_csrs: List[str] = []
-        for unit in certificates_relation.units:
-            requirer_relation_data = _load_relation_data(certificates_relation.data[unit])
-            requirer_csrs = requirer_relation_data.get("certificate_signing_requests", [])
-            list_of_csrs.extend(csr["certificate_signing_request"] for csr in requirer_csrs)
-        provider_certificates = provider_relation_data.get("certificates", [])
-        for certificate in provider_certificates:
-            if certificate["certificate_signing_request"] not in list_of_csrs:
-                self.on.certificate_revocation_request.emit(
-                    certificate=certificate["certificate"],
-                    certificate_signing_request=certificate["certificate_signing_request"],
-                    ca=certificate["ca"],
-                    chain=certificate["chain"],
-                )
-                self.remove_certificate(certificate=certificate["certificate"])
-
 
 class TLSCertificatesRequiresV1(Object):
     """TLS certificates requirer class to be instantiated by TLS certificates requirers."""
 
-    on = CertificatesRequirerCharmEvents()
-
     def __init__(
         self,
         charm: CharmBase,
         relationship_name: str,
-        expiry_notification_time: int = 168,
     ):
         """Generates/use private key and observes relation changed event.
 
         Args:
             charm: Charm object
             relationship_name: Juju relation name
-            expiry_notification_time (int): Time difference between now and expiry (in hours).
-                Used to trigger the CertificateExpiring event. Default: 7 days.
         """
         super().__init__(charm, relationship_name)
```

Comment:
Suggest to add deprecation log also for this class.
logger.warning(
            "This version of the `tls-certificates` library is deprecated. "
            "Please use the latest version."
        )

Reply:
Good catch, I missed it. It's gone now.